zanata-war/src/main/java/org/zanata/action/ProjectHomeAction.java

```diff
@@ -438,4 +486,383 @@ public void resetPageData() {
+    }
+
+    private final Function<Map.Entry<HLocale, Collection<LocaleRole>>, String>
+            TO_LOCALE_ROLES_DISPLAY_STRING =
+            new Function<Map.Entry<HLocale, Collection<LocaleRole>>, String>() {
+                @Nullable @Override public String apply(
+                        @Nullable
+                        Map.Entry<HLocale, Collection<LocaleRole>> entry) {
+                    final String localeName = entry.getKey().retrieveDisplayName();
+
+                    final Collection<String> roleNames =
+                            Collections2.transform(entry.getValue(), TO_DISPLAY_NAME);
+
+                    return localeName + " " + Joiner.on(", ").join(roleNames);
+                }
+            };
+
+    private final Function<LocaleRole, String> TO_DISPLAY_NAME =
+            new Function<LocaleRole, String>() {
+                @Nullable @Override
+                public String apply(
+                        @Nullable LocaleRole role) {
+                    return localeRoleDisplayName(role);
+                }
+            };
+
+    public String projectRoleDisplayName(ProjectRole role) {
+        switch (role) {
+            case Maintainer:
+                return msgs.get("jsf.Maintainer");
+            case TranslationMaintainer:
+                return msgs.get("jsf.TranslationMaintainer");
+            default:
+                return "";
+        }
+    }
+
+    public String localeRoleDisplayName(LocaleRole role) {
+        switch (role) {
+            case Translator:
+                return msgs.get("jsf.Translator");
+            case Reviewer:
+                return msgs.get("jsf.Reviewer");
+            case Coordinator:
+                return msgs.get("jsf.Coordinator");
+            default:
+                return "";
+        }
+    }
+
+    private final class PeopleFilterComparator extends InMemoryListFilter<HPerson>
+        implements Comparator<HPerson> {
+
+        private final ProjectRolePredicate projectRolePredicate =
+                new ProjectRolePredicate();
+
+        private final ProjectLocalePredicate projectLocalePredicate =
+                new ProjectLocalePredicate();
+
+        private SortingType sortingType;
+
+        @Getter
+        @Setter
+        private boolean showMembersInGroup;
+
+        private List<HPerson> allMembers;
+
+        public PeopleFilterComparator(SortingType sortingType) {
+            this.sortingType = sortingType;
+        }
+
+        @Override
+        public int compare(HPerson o1, HPerson o2) {
+            SortingType.SortOption selectedSortOption =
+                sortingType.getSelectedSortOption();
+
+            if (!selectedSortOption.isAscending()) {
+                HPerson temp = o1;
+                o1 = o2;
+                o2 = temp;
+            }
+
+            if (selectedSortOption.equals(SortingType.SortOption.ROLE)) {
+                setShowMembersInGroup(true);
```

Comment:
@aeng could you please take a look at this method, there is something else with it that I did not understand (as well as Patrick's comment here).